exercises/queen-attack/tests/queen-attack.rs

```diff
@@ -3,79 +3,84 @@ extern crate queen_attack;
 use queen_attack::*;
 
 #[test]
-fn test_queen_creation_with_valid_position() {
-    let white_queen = Queen::new((2,4));
-    assert!(white_queen.is_ok());
+fn chess_position_on_the_board_is_ok() {
+    assert!(ChessPosition::new(2, 4).is_ok());
 }
 
 #[test]
 #[ignore]
-fn test_queen_creation_with_incorrect_positions() {
-    let white_queen = Queen::new((-1,2));
-    assert!(white_queen.is_err());
-
-    let white_queen = Queen::new((8,2));
-    assert!(white_queen.is_err());
+fn chess_position_off_the_board_is_err() {
+    assert!(ChessPosition::new(-1, 2).is_err());
+    assert!(ChessPosition::new(8, 2).is_err());
+    assert!(ChessPosition::new(5, -1).is_err());
+    assert!(ChessPosition::new(5, 8).is_err());
+}
 
-    let white_queen = Queen::new((5,-1));
-    assert!(white_queen.is_err());
+#[test]
+#[ignore]
+fn queen_is_created_with_a_ok_position() {
+    Queen::new(ChessPosition::new(2, 4).unwrap());
+}
 
-    let white_queen = Queen::new((5,8));
-    assert!(white_queen.is_err());
+#[test]
+#[ignore]
+#[should_panic(expected = "Invalid Position")]
+fn queen_panics_with_err_position() {
+    Queen::new(ChessPosition::new(-1, 4).unwrap());
 }
 
 #[test]
 #[ignore]
-fn test_can_not_attack() {
-    let white_queen = Queen::new((2,4)).unwrap();
-    let black_queen = Queen::new((6,6)).unwrap();
+fn queens_that_can_not_attack() {
+    let white_queen = Queen::new(ChessPosition::new(2, 4).unwrap());
+    let black_queen = Queen::new(ChessPosition::new(6, 6).unwrap());
     assert_eq!(false, white_queen.can_attack(&black_queen));
 }
 
 #[test]
 #[ignore]
-fn test_can_attack_on_same_rank() {
-    let white_queen = Queen::new((2,4)).unwrap();
-    let black_queen = Queen::new((2,6)).unwrap();
-    assert!(white_queen.can_attack(&black_queen));
+fn queens_on_the_same_rank_can_attack() {
+    let white_queen = Queen::new(ChessPosition::new(2, 4).unwrap());
+    let black_queen = Queen::new(ChessPosition::new(2, 6).unwrap());
+    assert_eq!(true, white_queen.can_attack(&black_queen));
 }
 
 #[test]
 #[ignore]
-fn test_can_attack_on_same_file() {
-    let white_queen = Queen::new((4,5)).unwrap();
-    let black_queen = Queen::new((3,5)).unwrap();
-    assert!(white_queen.can_attack(&black_queen));
+fn queens_on_the_same_file_can_attack() {
+    let white_queen = Queen::new(ChessPosition::new(4, 5).unwrap());
+    let black_queen = Queen::new(ChessPosition::new(3, 5).unwrap());
+    assert_eq!(true, white_queen.can_attack(&black_queen));
 }
 
 #[test]
 #[ignore]
-fn test_can_attack_on_first_diagonal() {
-    let white_queen = Queen::new((2,2)).unwrap();
-    let black_queen = Queen::new((0,4)).unwrap();
-    assert!(white_queen.can_attack(&black_queen));
+fn queens_on_the_same_diagonal_can_attack_one() {
```

Comment:
Hmm, while we are changing the test names, maybe we can take the opportunity to bikeshed. I could see these being `upper_right`, `upper_left`, etc. rather than just `one`, `two`, etc.

But I don't actually care, so you can pick whatever makes the most sense.